### PR TITLE
Add rich-text-react-renderer

### PR DESCRIPTION
**Pull request by connor-baer**
## Changes

- **Add `rich-text-react-renderer`** based on the [great work](https://github.com/mgmolisani/rich-text-react-renderer) by @mgmolisani 👏 , with a similar API to the existing `rich-text-html-renderer`.
- **Adapt `jest` and `rollup` configs** to work with `tsx`files. TypeScript requires the use of this extension for files that contain `jsx`.

## Todos

- [x] Make the tests actually work (`jest` is still struggling with `tsx` files)
- [x] Review the types (I'm not sure if I used the correct React types for the components) 
- [x] Update README with proper documentation (copied from `rich-text-html-renderer` so the examples need to be adapted)
- [x] Add link and description in the main README

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -4,18 +4,19 @@ import sourceMaps from 'rollup-plugin-sourcemaps';
 import typescript from 'rollup-plugin-typescript2';
 import json from 'rollup-plugin-json';
 
-export default outputFile => ({
-  input: 'src/index.ts',
+export default (outputFile, inputFile = 'src/index.ts', external) => ({
```
Comment (connor-baer) [inline rollup.config.js]: I wonder if there's a better way to extend/overwrite the default configuration. Perhaps the options should be an object? Or add a second `options` arg that is merged with the default config and can overwrite it?

File: packages/rich-text-react-renderer/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    const children: Next = nodes => nodeListToReactTree(nodes, { renderMark, renderNode });
+    if (!node.nodeType || !renderNode[node.nodeType]) {
+      // TODO: Figure what to return when passed an unrecognized node.
+      return null;
+    }
+    return renderNode[node.nodeType](node, children);
+  }
+}
+
+function appendKeyToValidElement(element: ReactNode, key: string | number): ReactNode {
```
Comment (sbezludny) [inline packages/rich-text-react-renderer/src/index.tsx]:
```suggestion
function appendKeyToValidElement(element: ReactNode, key: React.Key): ReactNode {
```

File: packages/rich-text-react-renderer/src/__test__/documents/ul.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import { Document } from '@contentful/rich-text-types';
+
+export default {
+  data: {},
+  content: [
```
Comment (sbezludny) [inline packages/rich-text-react-renderer/src/__test__/documents/ul.ts]: It seems that we need to find a better way of organising mocks. I will take of it once we merge this PR.

File: packages/rich-text-react-renderer/src/__test__/documents/embedded-entry.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { Document, BLOCKS } from '@contentful/rich-text-types';
+
+export default function(entry: Object) {
```
Comment (Khaledgarbaya) [inline packages/rich-text-react-renderer/src/__test__/documents/embedded-entry.ts]: I think we can make entry type more specific using an Entry type.
You can copy the type definition from the SDK [index.d.ts](https://github.com/contentful/contentful.js/blob/master/index.d.ts#L73-L77). WDYT?

File: packages/rich-text-react-renderer/src/__test__/documents/inline-entity.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import { Document, BLOCKS, INLINES } from '@contentful/rich-text-types';
+
+export default function inlineEntity(entry: Object, inlineType: INLINES) {
```
Comment (Khaledgarbaya) [inline packages/rich-text-react-renderer/src/__test__/documents/inline-entity.ts]: Same comment

File: rollup.config.js (hunk 1 above)
Comment (cribbles) [reply rollup.config.js]: The latter seems like the cleaner approach to me.

**Comment by cribbles** [comment]
Great job so far. I'll give it a final look once the doc updates are pushed up. Thanks!

File: packages/rich-text-react-renderer/src/index.tsx (hunk 2 above)
Comment (connor-baer) [reply packages/rich-text-react-renderer/src/index.tsx]: As @mgmolisani pointed out, this is the internal key that we assign, which is always a number.

File: packages/rich-text-react-renderer/src/__test__/documents/embedded-entry.ts (hunk 4 above)
Comment (connor-baer) [reply packages/rich-text-react-renderer/src/__test__/documents/embedded-entry.ts]: I tried to use the `Entry` type, however, it led to a bunch of issues: this isn't a full Contentful entry, as you can see [here](https://github.com/contentful/rich-text/pull/71/files#diff-c25141e0c89b52284671547a52d73c68R99). Considering that these are only used in the tests, I'd argue that we don't need to be that specific with the entry type.

Comment (sbezludny) [reply packages/rich-text-react-renderer/src/__test__/documents/embedded-entry.ts]: Good point about Entry type

```suggestion
export default function(entry: {sys: Contentful.Sys}) {
```

or even `Partial<Entry>` could work here.

Comment (sbezludny) [reply packages/rich-text-react-renderer/src/__test__/documents/embedded-entry.ts]: but I think we can take care of the mocks in one of the follow-ups

Comment (connor-baer) [reply packages/rich-text-react-renderer/src/__test__/documents/embedded-entry.ts]: Do you wanna tackle this when you [reorganise the mocks](https://github.com/contentful/rich-text/pull/71#discussion_r254430417)? These documents are copied one for one from the `rich-text-html-renderer`, including types. This should be updated there as well.

**Comment by sbezludny** [comment]
@mgmolisani 

> Should I submit a PR to correct this and align the jest types libs?

Yes, please.

**Comment by mgmolisani** [comment]
@sbezludny
> Yes, please.

New PR created for this followup. #74

**Comment by connor-baer** [comment]
@sbezludny @cribbles @Khaledgarbaya We're ready to merge and release this. Can you do a final review, please?

**Comment by mirshko** [comment]
love you peeps so much <3

**Comment by Khaledgarbaya** [comment]
Hey @connor-baer and @mgmolisani.
Can you ping me at `khaled[at]contentful.com`. We would like to send some nice swags your ways.

Cheers,
Khaled

**Pull request merged**